app/controllers/api/v1/teachers_controller.rb

```diff
@@ -44,6 +44,7 @@ def create
         current_user.email = session['omniauth_email']
         current_user.save!
         session['omniauth_email'] = nil
+        current_user.email_subscribed = params[:email_subscribed]
```

Comment:
You should do this before the current_user.save! call.
Otherwise this won't be preserved. It isn't strictly necessary, but it would be confusing if it isn't saved.

Reply:
Ah, I see. Thanks. I'll make that change.